torch/csrc/cuda/nccl.h

```diff
@@ -6,11 +6,18 @@
 #include <c10/cuda/CUDACachingAllocator.h>
 #include <c10/util/Optional.h>
 
-#include <nccl.h>
-
 #include <cstddef>
 #include <vector>
 
+enum ncclDataType_t : unsigned int;
+enum ncclResult_t : unsigned int;
+enum ncclRedOp_t : unsigned int;
+struct ncclComm_t;
```

Comment:
This forward declaration is wrong, isn't it?
At least according to https://github.com/NVIDIA/nccl/blob/5949d96f36d050e59d05872f8bbffd2549318e95/src/nccl.h.in#L26 it should be something like
```suggestion
typedef struct ncclComm* ncclComm_t;
```